### PR TITLE
Backport to v2.3.x fix for correctly parsing enum uriParameters

### PR DESCRIPTION
**Pull request by yannickroger**
Hello,
I am working on a project that is still running RAML 0.8 and we ran into a bug that has been fixed in a later version.

Here is a cherry pick of the commit on based on 2.3.4. Unfortunately there is not branch to point this pull request to.

Could you create a 2.3.5 tag based on this branch please.

### CODE REVIEW

File: src/Resource.php [hunk 1]
```diff
@@ -221,7 +221,8 @@ public function matchesUri($uri)
 
         $regexUri = preg_replace('/\/{.*}/U', '\/([^/]+)', $regexUri);
         $regexUri = preg_replace('/\/~{.*}/U', '\/([^/]*)', $regexUri);
-        $regexUri =  '|^' . $regexUri . '$|';
+        // начало и конец регулярки - символ, который гарантированно не встретится
```
Comment (martin-georgiev) [inline src/Resource.php]:
```suggestion
```

The repository is developed and maintained in English. When ports from other forks take place we should avoid blinding transferring non-English comments. It's fine to translate them though.

**Comment by martin-georgiev** [comment]
@yannickroger Can you please accept the code suggestion?
I've updated the target branch to `v2.x` and would tag a new release once this is merged.

**Comment by yannickroger** [comment]
@martin-georgiev  done. Thanks a lot.

**Comment by martin-georgiev** [comment]
Released as v2.3.5

**Comment by yannickroger** [comment]
Thanks again @martin-georgiev

**Comment by yannickroger** [comment]
@martin-georgiev The version is not showing up on packagist (the ones made yesterday also) https://packagist.org/packages/raml-org/raml-php-parser

**Comment by martin-georgiev** [comment]
> @martin-georgiev The version is not showing up on packagist (the ones made yesterday also) https://packagist.org/packages/raml-org/raml-php-parser

Can you check something is not being cached on your end?
Packagist is up-to-date for me (see below).
Screenshot 2019-04-12 at 17 22 24

**Comment by yannickroger** [comment]
It all works now (composer update and packagist site). I don't think it was cached as it is the first time I go on packagist with this computer and I am on a home network.

I guess it was just processed a few minutes ago. Sorry for the noise.

**Comment by martin-georgiev** [comment]
No harm was done :). Hopefully, it will work for you now.

**Pull request merged**